wingtips-zipkin2/src/test/java/com/nike/wingtips/zipkin2/util/WingtipsToZipkinSpanConverterDefaultImplTest.java

```diff
@@ -177,6 +178,40 @@ public void convertWingtipsSpanToZipkinSpan_throws_IllegalArgumentException_when
         assertThat(ex).isInstanceOf(IllegalArgumentException.class);
     }
 
+    @DataProvider(value = {
+            "123e4567-e89b-12d3-a456-426655440000  ",  // UUID format (hyphens and also >32 chars)
+            "48485w3953Zz6124"  // not lower-case hex
+    }, splitBy = "\\|")
+    @Test
+    @SuppressWarnings("UnnecessaryLocalVariable")
+    public void convertWingtipsSpanToZipkinSpan_works_as_expected_for_invalid_traceId_format(final String badHexString) {
+        // given
+        String badTraceId = badHexString;
+        String spanId = "48485a3953bb6124";
+        String spanName = UUID.randomUUID().toString();
+        long startTimeEpochMicros = Math.abs(random.nextLong());
+        long durationNanos = Math.abs(random.nextLong());
+        final Endpoint zipkinEndpoint = Endpoint.newBuilder().serviceName(UUID.randomUUID().toString()).build();
+        final Span wingtipsSpan = Span.newBuilder(spanName, SpanPurpose.CLIENT)
+                .withTraceId(badTraceId)
+                .withSpanId(spanId)
+                .withSpanStartTimeEpochMicros(startTimeEpochMicros)
+                .withDurationNanos(durationNanos)
+                .build();
+
+        // when
+        zipkin2.Span zipkinSpan = impl.convertWingtipsSpanToZipkinSpan(wingtipsSpan, zipkinEndpoint);
+
+        // then
+        assertThat(zipkinSpan.traceId()).matches("^[a-f0-9]+$");    // lower hex
```

Comment:
Since it's a hash, it's deterministic and predictable. Rather than asserting that's the result is simply lowerhex and has the allowed number of characters, the test's `@DataProvider` test cases should include the exact expected result. The assertion should be `assertThat(zipkinSpan.traceId()).isEqualTo(expectedSanitizedTraceIdResult)`, with `expectedSanitizedTraceIdResult` coming from the data provider test case. Otherwise you could just generate a random valid-format ID and it would pass this test.